test/SlowTests/Issues/RavenDB-17597.cs

```diff
@@ -0,0 +1,172 @@
+using Raven.Client.Exceptions.Database;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Raven.Server;
+using Raven.Server.Commercial;
+using Raven.Server.Config;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Indexes;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17597 : ClusterTestBase
+    {
+        public RavenDB_17597(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ModifyTaskIndexTest()
+        {
+            using var server = GetNewServer(new ServerCreationOptions { RunInMemory = false, });
+            using var store = GetDocumentStore(new Options { RunInMemory = false, Server = server });
+            {
+                // Prepare Server For Test
+                for(int i = 0; i < 5; i++)
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new Item
+                    {
+                        Num = 5
+                    });
+                    await session.SaveChangesAsync();
+                }
+
+                // Wait for indexing in first node
+                var index = new Index_ItemsByNum();
+                index.Execute(store);
+                Indexes.WaitForIndexing(store);
+
+                // Test
+                CompactSettings settings = new CompactSettings { DatabaseName = store.Database, Documents = true, Indexes = new[] { index.IndexName } };
+
+                //Modify index
+                try
+                {
+                    var index2 = new Index_ItemsByNum2();
+                    index2.Execute(store);
+                    Indexes.WaitForIndexing(store);
+                }
+                catch(InvalidOperationException) { }
+                string[] indexNames1 = store.Maintenance.Send(new GetIndexNamesOperation(0, 10));
+                Assert.NotNull(indexNames1);
+                Assert.Equal(2, indexNames1.Length);
+                Assert.Contains("Items/ByNum", indexNames1);
+                Assert.Contains("ReplacementOf/Items/ByNum", indexNames1);
+                
+                //restart server
+                var result = await DisposeServerAndWaitForFinishOfDisposalAsync(server);
+                using var newServer = GetNewServer(new ServerCreationOptions
+                {
+                    DeletePrevious = false,
+                    RunInMemory = false,
+                    DataDirectory = result.DataDirectory,
+                    CustomSettings = new Dictionary<string, string> { [RavenConfiguration.GetKey(x => x.Core.ServerUrls)] = result.Url }
+                });
+                await Task.Delay(TimeSpan.FromSeconds(10)); // wait for ExecuteForIndexes call in IndexStore.HandleDatabaseRecordChange method.
```

Comment:
let's try to avoid those delays.
1. use `newServer.InitializationCompleted`
2. wait on the database it self to completed applied the last raft index, by accessing `RachisLogIndexNotifications` of the database

Reply:
Done